### PR TITLE
fix exitcode

### PR DESCRIPTION
**Pull request by Yancey1989**

### CODE REVIEW

File: scripts/python/tao_build.py [hunk 1]
```diff
@@ -900,10 +900,13 @@ def main():
...
+    exit_code = 0
     try:
         main()
     except Exception as e:
         import traceback
         traceback.print_exc()
+        exit_code = -1
     finally:
         stage_time.report()
+    exit(exit_code)
```
Comment (linearhit) [inline scripts/python/tao_build.py:912]: what's the difference if we don't use "try" at all?

Comment (Yancey1989) [reply scripts/python/tao_build.py:912]: we can print the report at "finally", and also print the traceback at failed, it's easier to debug.

Comment (qiuxiafei) [reply scripts/python/tao_build.py:912]: Rethrowing the exception in `catch` block could also result to non-zero exit code.

Comment (Yancey1989) [reply scripts/python/tao_build.py]: > Rethrowing the exception in catch block could also result to non-zero exit code.

@qiuxiafei thanks! Done.

**Review by wyzero — APPROVED** [review]
LGTM

**Pull request merged**